src/webgpu/web_platform/reftests/canvas_composite_alpha.html.ts

```diff
@@ -0,0 +1,129 @@
+import { assert, unreachable } from '../../../common/util/util.js';
+
+import { runRefTest } from './gpu_ref_test.js';
+
+// <canvas> element from html page
+declare const cvs: HTMLCanvasElement;
+
+export function run(format: GPUTextureFormat, compositingAlphaMode: GPUCanvasCompositingAlphaMode) {
+  runRefTest(async t => {
+    const ctx = cvs.getContext('gpupresent');
+    assert(ctx !== null, 'Failed to get WebGPU context from canvas');
+
+    switch (format) {
+      case 'bgra8unorm':
+      case 'bgra8unorm-srgb':
+      case 'rgba8unorm':
+      case 'rgba8unorm-srgb':
+        break;
+      default:
+        unreachable();
+    }
+
+    const a = compositingAlphaMode === 'opaque' ? (1.0).toFixed(1) : (0.5).toFixed(1);
```

Comment:
The `'opaque'` compositingAlphaMode is supposed to ignore the contents of the alpha channel and display the texture as if alpha were cleared to 1. So I don't think this would do anything.

Actually it would be best to test multiple different alpha values (e.g. on different pixels, output  = 0.0, 0.5, or 1.0) just to make sure all cases behave the same way.

Reply:
There's blending inside canvas happening here. We are having 4 draw calls so each draw result still blends with each other inside the canvas. This is to mimic the behavior of canvas 2d ref to achieve the same in-canvas result. (although what we are really testing is canvas blending with other elements on page)